frontend/src/lib/k8s/index.ts

```diff
@@ -140,6 +140,17 @@ export function useClusterGroup(): string[] {
   return clusterGroup;
 }
 
+/**
+ * Use the cluster name from the URL query parameters if it's there.
+ *
+ * @returns the cluster name from the URL. If no cluster is defined in the URL, undefined is returned.
+ */
+export function useClusterFromURLVar(): string | undefined {
```

Comment:
I think we should be very explicit about what this means, because we already have a cluster 'variable' in the url /c/cluster/. To disambiguate I think this should be named something like `useClusterFromQueryParam`

Reply:
What do you mean, never mind?

Reply:
I was thinking “query” is maybe overloaded with react query stuff.  But maybe it’s ok.

useClusterFromSearchParam is also not quite right making it sound like it’s from a search.

useClusterFromURLQuery?
useURLQueryCluster?

Reply:
well there's a separate discussion about the url format, so this comment may not be relevant

Reply:
useURLQueryCluster sounds good
hm or maybe an approach of naming it for what it is instead of where it's stored, something like `useDetailsSingleCluster`?

Reply:
Yeah maybe useDetailsSingleCluster is the least misdirecting/confusing.

Reply:
> well there's a separate discussion about the url format, so this comment may not be relevant, let me rephrase it better

Can you please let us know which proposal you’d prefer and why?

Reply:
posted it in a separate comment